purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/trustedentitlements/TrustedEntitlementsInformationalModeIntegrationTest.kt

```diff
@@ -0,0 +1,162 @@
+package com.revenuecat.purchases.trustedentitlements
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.BasePurchasesIntegrationTest
+import com.revenuecat.purchases.CacheFetchPolicy
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.EntitlementVerificationMode
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.PurchaseParams
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.VerificationResult
+import com.revenuecat.purchases.factories.StoreProductFactory
+import com.revenuecat.purchases.factories.StoreTransactionFactory
+import com.revenuecat.purchases.forceSigningErrors
+import com.revenuecat.purchases.getCustomerInfoWith
+import com.revenuecat.purchases.helpers.mockQueryProductDetails
+import com.revenuecat.purchases.purchaseWith
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+@RunWith(AndroidJUnit4::class)
+class TrustedEntitlementsInformationalModeIntegrationTest : BasePurchasesIntegrationTest() {
+
+    @Before
+    fun setup() {
+        ensureBlockFinishes { latch ->
+            setUpTest(entitlementVerificationMode = EntitlementVerificationMode.INFORMATIONAL) {
+                latch.countDown()
+            }
+        }
+    }
+
+    @Test
+    fun initialCustomerInfoIsVerified() {
+        var receivedCustomerInfo: CustomerInfo? = null
+        ensureBlockFinishes { latch ->
```

Comment:
Can't wait to see these with coroutines 😅

Reply:
Pretty straightforward and neat 💅 https://github.com/RevenueCat/purchases-android/pull/1077